### PR TITLE
Add export for HotTableClass in React wrapper

### PR DESCRIPTION
**Pull request by evanSe**
### Context
With the refactor we did around HotTable and HotTableClass for `useId` we did not export `HotTableClass`, this pull request simply exports `HotTableClass`. 

### How has this been tested?
Locally

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/10736

### Affected project(s):
- [x] `@handsontable/react`

### CODE REVIEW

File: .changelogs/10736.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "issuesOrigin": "public",
+  "title": "Export HotTableClass from React wrapper\"",
```
Comment (jansiegel) [inline .changelogs/10736.json:3]: I think there's a typo here with `\"`?

**Review by jansiegel — APPROVED** [review]

**Review by budnix — APPROVED** [review]

File: .changelogs/10736.json (hunk 1 above)
Comment (evanSe) [inline .changelogs/10736.json:3]:
```suggestion
  "title": "Export HotTableClass from React wrapper",
```

Comment (evanSe) [reply .changelogs/10736.json]: Yes, thanks for pointing out

**Review by jansiegel — APPROVED** [review]

**Comment by NOtherDev** [comment]
@evanSe Can I ask you to merge `develop` to `feature/improved-react` as this (and few others) will be useful there?

**Comment by evanSe** [comment]
> feature/improved-react

Done

**Pull request merged**